test/unit-test/test-Model.js

```diff
@@ -6836,6 +6836,596 @@ describe('Unit Test/Model Test', function() {
+      await assertThrowsAsync(async() => {
+        await execution.startCompute();
+      });
+    });
+
+    it('raise error when only one input that is 4-D tensor as TENSOR_FLOAT32 type for "BATCH_TO_SPACE_ND" operation', async function() {
+      let model = await nn.createModel(options);
+      let input = {type: nn.TENSOR_FLOAT32, dimensions: [4, 1, 1, 1]};
+      let output = {type: nn.TENSOR_FLOAT32, dimensions: [1, 2, 2, 1]};
+
+      model.addOperand(input);
+      model.addOperand(output);
+
+      model.addOperation(nn.BATCH_TO_SPACE_ND, [0], [1]);
+      model.identifyInputsAndOutputs([0], [1]);
+      await model.finish();
+
+      let compilation = await model.createCompilation();
+      compilation.setPreference(getPreferenceCode(options.prefer));
+      await compilation.finish();
+
+      let execution = await compilation.createExecution();
+      execution.setInput(0, new Float32Array([1, 2, 3, 4]));
+      execution.setOutput(0, new Float32Array(product(output.dimensions)));
+      await assertThrowsAsync(async() => {
+        await execution.startCompute();
+      });
+    });
+
+    it('raise error when setting 0 input for "BATCH_TO_SPACE_ND" operation', async function() {
+      let model = await nn.createModel(options);
+      let output = {type: nn.TENSOR_FLOAT32, dimensions: [1, 2, 2, 1]};
+
+      model.addOperand(output);
+
+      model.addOperation(nn.BATCH_TO_SPACE_ND, [], [0]);
+      model.identifyInputsAndOutputs([], [0]);
+      await model.finish();
+
+      let compilation = await model.createCompilation();
+      compilation.setPreference(getPreferenceCode(options.prefer));
+      await compilation.finish();
+
+      let execution = await compilation.createExecution();
+      execution.setOutput(0, new Float32Array(product(output.dimensions)));
+      await assertThrowsAsync(async() => {
+        await execution.startCompute();
+      });
+    });
+
+    it('raise error when setting 0 output for "BATCH_TO_SPACE_ND" operation', async function() {
+      let model = await nn.createModel(options);
+      let input0 = {type: nn.TENSOR_FLOAT32, dimensions: [4, 1, 1, 1]};
+      let input1 = {type: nn.TENSOR_INT32, dimensions: [2]};
+
+      model.addOperand(input0);
+      model.addOperand(input1);
+
+      model.setOperandValue(1, new Int32Array([2, 2]));
+      model.addOperation(nn.BATCH_TO_SPACE_ND, [0, 1], []);
+      model.identifyInputsAndOutputs([0], []);
+      await model.finish();
+
+      let compilation = await model.createCompilation();
+      compilation.setPreference(getPreferenceCode(options.prefer));
+      await compilation.finish();
+
+      let execution = await compilation.createExecution();
+      execution.setInput(0, new Float32Array([1, 2, 3, 4]));
+      await assertThrowsAsync(async() => {
+        await execution.startCompute();
+      });
+    });
+
+    it('raise error when the block value < 1 for "BATCH_TO_SPACE_ND" operation', async function() {
```

Comment:
This test case is for input1 isn't of TENSOR_INT32 type, please add one test case with  input1 as TENSOR_INT32 tensor and its' value < 1.